src/callbacks/checkpoint.cpp

```diff
@@ -489,28 +425,141 @@ bool checkpoint::restart(model *m) {
   // one.
   // Then setup the model with the proper one
   auto& c = static_cast<sgd_execution_context&>(m->get_execution_context());
+  return open_latest_checkpoint(
+    *(m->get_comm()),
+    "Restarting",
+    get_active_trainer().get_name(),
+    get_active_training_algorithm().get_name(),
+    [&m, &c](persist& p_ref) {
+      return c.get_trainer().load_from_checkpoint_distributed(*m, c);
+    },
+    [&m, &c](persist& p_ref) {
+      return c.get_trainer().load_from_checkpoint_distributed(*m, c);
+    });
```

Comment:
Is this implementation change intended?
```suggestion
  return open_latest_checkpoint(
    *(m->get_comm()),
    "Restarting",
    get_active_trainer().get_name(),
    get_active_training_algorithm().get_name(),
    [&m, &c](persist& p_ref) {
      return c.get_trainer().load_from_checkpoint_shared(*m, c);
    },
    [&m, &c](persist& p_ref) {
      return c.get_trainer().load_from_checkpoint_distributed(*m, c);
    });
```

Reply:
Copy-paste error, thanks!